esp8266/modesp.c

```diff
@@ -530,6 +530,21 @@ STATIC mp_obj_t esp_osdebug(mp_obj_t val) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp_osdebug_obj, esp_osdebug);
 
+STATIC mp_obj_t esp_save_config() {
+    if ((wifi_get_opmode() & STATION_MODE) == 1) {
+        struct station_config config = {{0}};
+        wifi_station_get_config(&config);
```

Comment:
perhaps, better would be to use `error_check()`
`error_check(wifi_station_get_config(&config), "can't get STA config");`